File: Source/DotNET/Model/Forms/PopulateFromItem.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using Cratis.Scene.Model.Common;

namespace Cratis.Scene.Model.Forms;

/// <summary>
/// Populates a <see cref="Form"/>'s initial values from an item already in scope (e.g. a row selected in
/// a table on the same screen), rather than a fresh query.
/// </summary>
/// <param name="ItemBinding">The binding to the item already in scope.</param>
public record PopulateFromItem(BindingExpression ItemBinding) : PopulateSource;

File: Source/JavaScript/model/forms/PopulateFromItem.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { PopulateSource } from './PopulateSource';
import { BindingExpression } from '../common';

/**
 * Populates a {@link Form}'s initial values from an item already in scope (e.g. a row selected in a
 * table on the same screen), rather than a fresh query.
 */
export interface PopulateFromItem extends PopulateSource {
    itemBinding: BindingExpression;
}

export const PopulateFromItemPropertyNames: (keyof PopulateFromItem)[] = ['itemBinding'];
